src/_P014_SI70xx.ino

```diff
@@ -0,0 +1,198 @@
+      options[0]      = F("Temp 14 bits / RH 12 bits");
+      optionValues[1] = SI70xx_RESOLUTION_13T_10RH;
+      options[1]      = F("Temp 13 bits / RH 10 bits");
+      optionValues[2] = SI70xx_RESOLUTION_12T_08RH;
+      options[2]      = F("Temp 12 bits / RH  8 bits");
+      optionValues[3] = SI70xx_RESOLUTION_11T_11RH;
+      options[3]      = F("Temp 11 bits / RH 11 bits");
+      addFormSelector(F("Resolution"), F("p014_res"), P014_RESOLUTION_OPTIONS, options, optionValues, choice);
+
+      addFormNumericBox("ADC Filter Power", F("p014_filter"), P014_FILTER_POWER, 0, 4);
+
+      // addUnit(F("bits"));
+
+      success = true;
+      break;
+    }
+
+    case PLUGIN_WEBFORM_SAVE:
+    {
+      P014_I2C_ADDRESS  = getFormItemInt(F("i2c_addr"));
+      P014_RESOLUTION   = getFormItemInt(F("p014_res"));
+      P014_FILTER_POWER = getFormItemInt(F("p014_filter"));
+        
+      // Force device setup next time
+      P014_data_struct *P014_data = static_cast<P014_data_struct *>(getPluginTaskData(event->TaskIndex));
+      if (nullptr != P014_data) {
+        P014_data->state = P014_state::Uninitialized;
+      }
+
+      success = true;
+      break;
+    }
+
+
+    case PLUGIN_INIT:
+    {
+      initPluginTaskData(event->TaskIndex, new (std::nothrow) P014_data_struct());
+      P014_data_struct *P014_data = static_cast<P014_data_struct *>(getPluginTaskData(event->TaskIndex));
+
+      if (nullptr == P014_data) {
+        return success;
+      }
+      if (P014_data->init(P014_I2C_ADDRESS, P014_RESOLUTION)) {
+        success = true;
+      }else{
+        UserVar[event->BaseVarIndex]     = NAN;
+        UserVar[event->BaseVarIndex + 1] = NAN;
+        UserVar[event->BaseVarIndex + 2] = NAN;
+      }
+      break;
+    }
+
+
+    case PLUGIN_READ:
+    {
+      P014_data_struct *P014_data = static_cast<P014_data_struct *>(getPluginTaskData(event->TaskIndex));
+
+      if (nullptr != P014_data) {
+        if (P014_data->state == P014_state::Error){
+          UserVar[event->BaseVarIndex]     = NAN;
+          UserVar[event->BaseVarIndex + 1] = NAN;
+          UserVar[event->BaseVarIndex + 2] = NAN;
+
+          addLog(LOG_LEVEL_ERROR,F("SI70xx: in Error!"));
+        
+          return false;
+        }
+        
+        if (P014_data->state != P014_state::New_Values_Available) {
+          P014_data->update(P014_I2C_ADDRESS,P014_RESOLUTION,P014_FILTER_POWER); //run the state machine
+          Scheduler.schedule_task_device_timer(event->TaskIndex, millis() + SI70xx_DELAY);//keep going until we have New_values_available
+
+          return false; //we are not ready to read the values
+        }
+        
+        UserVar[event->BaseVarIndex]     = P014_data->temperature / 100.0f;
+        UserVar[event->BaseVarIndex + 1] = P014_data->humidity / 10.0f;
+        if (P014_data->chip_id == CHIP_ID_SI7013){
+          UserVar[event->BaseVarIndex + 2] = (P014_data->adc) >> P014_FILTER_POWER;
```

Comment:
The value count should be variable as the older implementation used 2 task values.
I get it that there is another output value possible with the SI7013, but now this 3rd value will always be NaN on existing setups.
Some controllers do not send data (any data) if there are NaN values.

For the analog value to be added, you should make this configurable.
See for example the Dallas plugin, or sysinfo or Dummy. (and others)
These can report the actual nr of task values which also is used on preparing data to send to controllers.

Also now this task may send upto 3 events where the last one is never useful on SI7021 sensors.

Reply:
ok. I think I pushed some time back an update that would not send events if the name of the field is set to empty.
So it could be easily handled that way.
I will look into implementing that dropdown too.

Reply:
> ok. I think I pushed some time back an update that would not send events if the name of the field is set to empty.

That's also something that's not desired.
Having an empty name is intended to be using the default name.
This is particularly useful when a plugin allows to select several different options to set as task values.
For example the sysinfo plugin.

I recently also added a caching mechanism to speed up execution on mainly ESP32 to get the info from the ExtraTaskSettings. This also uses this idea to have a default taskvaluename.

Reply:
So how should be handled if someone just want temperature or humidity alone ?
Or in this SI7013 just the ADC output to be sent?

Reply:
You can use a Dummy device as an intermediate, and a short rule to move the desired value of the sensor to the Dummy:
```
on SI7013#Humidity do
  TaskValueSet,Outgoing,Humidity,%eventvalue1% // Assumes a Dummy device named Outgoing set to Single and value Humidity, configured to use a controller
endon
```

Reply:
yes. but that would be a hack vs supporting it in the plugin itself.
I also think the approach of TEMP_BARO which skips second value is way more confusing.
I think makes no sense to send empty names for values thus having it empty  is a short term approach.
On the long term maybe having a bitwise mask and enable each value for sending to the controller.

Reply:
Something like I did here: https://github.com/letscontrolit/ESPEasy/pull/4011 / [Energy (DC) - INA3221](https://espeasy.readthedocs.io/en/latest/Plugin/P132.html#p132-page)

Reply:
You can check the Accu Energy plugin, or GPS, or sysinfo to see how you can select exactly which one you want as task value and also on what task value index.

Reply:
yes that solves the problem by adding a lot of code. where you guys were complaining that I had same line in two places.